### PR TITLE
Fixed errors on pubsub_publish

### PR DESCRIPTION
**Pull request by william-rgt**
Changes related to:
https://cloudwalk.atlassian.net/browse/CCT-147
https://cloudwalk.atlassian.net/browse/CCT-280
https://cloudwalk.atlassian.net/browse/CCT-285
https://cloudwalk.atlassian.net/browse/CCT-284
https://cloudwalk.atlassian.net/browse/CCT-283
https://cloudwalk.atlassian.net/browse/CCT-282
https://cloudwalk.atlassian.net/browse/CCT-271
https://cloudwalk.atlassian.net/browse/CCT-126
https://cloudwalk.atlassian.net/browse/CCT-295
https://cloudwalk.atlassian.net/browse/CCT-281
https://cloudwalk.atlassian.net/browse/CCT-142

### CODE REVIEW

File: src/thread_scheduler.c [hunk 1]
```diff
@@ -704,15 +704,13 @@ mrb_thread_pub_sub_s__publish(mrb_state *mrb, mrb_value self)
     } else {
       len = pubsub_publish(RSTRING_PTR(buf), RSTRING_LEN(buf), -1);
     }
-    if (len > 0) return_value = mrb_true_value();
   }
-
-  return_value = mrb_false_value();
-
-  TRACE("return");
```
Comment (mauriciospinardi-cloudwalk) [inline src/thread_scheduler.c:712]: Either remove both `TRACE("return");` and `TRACE_FUNCTION();` or neither. They work as a pair.

File: src/thread_scheduler.c [hunk 2]
```diff
@@ -374,9 +374,9 @@ pubsub_publish(char *buf, int len, int avoid_id)
 {
   int id = 0, ret = 0;
 
-  while (id < 10 && conn_thread_events_marker[id])
+  while (id < 10)
   {
-    if (id != avoid_id)
+    if (conn_thread_events_marker[id] && id == avoid_id)
```
Comment (mauriciospinardi-cloudwalk) [inline src/thread_scheduler.c:379]: The conditional change from `!= avoid_id` to `== avoid_id` changes the core function intents.
If that change is to be kept, I would suggest renaming `avoid_id` to something like `target` (or `targeted_id`, etc.), in an attempt to clarify the meaning of the argument.

**Review by mauriciospinardi-cloudwalk — CHANGES_REQUESTED** [review]
Please, check standalone comments.

**Comment by william-rgt** [comment]
@mauriciospinardi-cloudwalk Fixed!

**Review by mauriciospinardi-cloudwalk — APPROVED** [review]

**Pull request merged**